### ISSUE TITLE
function constructor: /extern words are not checked against formal arguments

### ISSUE THREAD

**Issue by meijeru** [red/red#324]
`f: function [a [integer!] /extern [a]][a: 1]` is compiled without protest

**Comment by rebolek** [comment]
0.6.1 can compile this function without any error raised.

**Comment by hiiamboris** [comment]
0.6.4:
compiles `/extern [a]` without an error, `/extern a` reports an error
interprets both `/extern [a]` and `/extern a` without an error

**Comment by meijeru** [comment]
But... `/extern [a]` is not recognized, resulting in `a` being local!, whereas `/extern a` has the desired effect.

**Comment by greggirwin** [comment]
Do we have a spec that says `/extern [a]` is supported syntax?

**Comment by hiiamboris** [comment]
The point is to report an error in both cases. Which is doesn't.

**Comment by greggirwin** [comment]
Agreed. I thought you meant it should be supported, or that it was an open question.

**Comment by 9214** [comment]
`/extern` use-cases are pretty rare in my experience, typically for escaping 1 or 2 words. With that said, I don't think that it should be overburdened with `[...]` syntax. OTOH, existing Rebol syntax confuses new users, who think that it's a refinement like e.g. `/local`.

**Comment by dockimbel** [comment]
> OTOH, existing Rebol syntax confuses new users, who think that it's a refinement like e.g. /local.

That's a good argument for keeping new users away from `function`, and make them use `func` only until they master the basic concepts.

**Comment by endo64** [comment]
Keeping `/extern` consistent with `/local` looks better to me, we don't need much speed on interpreter since `function`s with `/extern` is a rare case.
